cmd/meshregistrator/service_registration.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"hash/fnv"
+	"strconv"
+	"sync"
+)
+
+type ServiceRegistrationsList struct {
+	sync.Mutex
+	IsDirty bool
+	Items   []ServiceRegistration
+}
+
+type ServiceRegistration struct {
+	Service      string
+	Instance     string
+	PodNs        string
+	PodName      string
+	PodNode      string
+	PodIP        string
+	Port         int32
+	Registration string
+}
+
+func (s *ServiceRegistration) Hash() uint64 {
+	h := fnv.New64a()
+	h.Write([]byte(s.Service))
+	h.Write([]byte(s.Instance))
+	h.Write([]byte(s.PodNs))
+	h.Write([]byte(s.PodName))
+	h.Write([]byte(s.PodNode))
+	h.Write([]byte([]byte(strconv.Itoa(int(s.Port)))))
+	h.Write([]byte(s.PodIP))
+	h.Write([]byte(s.Registration))
+	return h.Sum64()
+}
+
+// compare registration slices ignoring order
+func registrationsEqual(x, y []ServiceRegistration) bool {
```

Comment:
is the performance win from hashing really worthwhile (as opposed to just comparing each field in the ServiceRegistration)? I'm almost certain at somepoint we'll be debugging why something did/didn't change and we'll want to see which of the fields being compared was different and have to write some code to assist in debugging

Reply:
this is not actually about performance, just a set-equality algorithm, but since go doesn't have generics i need to have this =/ naive implementation is potentially quadratic